### PR TITLE
Added info on long life tokens

### PR DESCRIPTION
**Pull request by mburke5678**
https://trello.com/c/bXuDfUNt/655-document-security-as-a-cluster-administrator-i-can-configure-openshift-to-issue-long-lived-api-tokens-to-specific-oauth-clients

### CODE REVIEW

**Comment by mburke5678** [comment]
@liggitt  PTAL

File: architecture/additional_concepts/other_api_objects.adoc [hunk 1]
```diff
@@ -117,20 +117,24 @@ metadata:
   selflink: "/osapi/v1/oAuthClients/openshift-web-console"
   resourceVersion: "1"
   creationTimestamp: "2015-01-01T01:01:01Z"
-respondWithChallenges: false <2>
-secret: "45e27750-a8aa-11e4-b2ea-3c970e4b7ffe" <3>
+  accessTokenMaxAgeSeconds: "172800" <2>
```
Comment (liggitt) [inline architecture/additional_concepts/other_api_objects.adoc]: don't make the default override ... if you want to include the field, set it to nil in the example, and document the meaning of `0` and non-zero values

Comment (liggitt) [reply architecture/additional_concepts/other_api_objects.adoc]: also, this is a number, not a string

**Comment by mburke5678** [comment]
@liggitt PTAL.

File: architecture/additional_concepts/other_api_objects.adoc [hunk 2]
```diff
@@ -111,26 +111,33 @@ The following `*OAuthClient*` objects are automatically created:
 [source,yaml]
 ----
 kind: "OAuthClient"
+accessTokenMaxAgeSeconds:  <1>
```
Comment (liggitt) [inline architecture/additional_concepts/other_api_objects.adoc]: `accessTokenMaxAgeSeconds: null`

**Review by liggitt — COMMENTED** [review]
Lgtm

**Comment by mburke5678** [comment]
@stuartchuan PTAL

File: architecture/additional_concepts/other_api_objects.adoc [hunk 3]
```diff
@@ -111,30 +111,43 @@ The following `*OAuthClient*` objects are automatically created:
...
 
+[[accessTokenMaxAgeSeconds]]
+The `accessTokenMaxAgeSeconds` value overrides the 
+xref:../../install_config/configuring_authentication.adoc#token-options[default `accessTokenMaxAgeSeconds` value] in the master configuration file 
+for specific OAuth clients. Setting this value for a client allows for long-lived access tokens for a client 
+without affecting the lifetime for other clients. 
+
+* If `null`, the default value in the master configuration file is used. 
+* If set to `0`, the token will not expire. 
+* If set to a value greater than `0`, tokens issued for that client are given the specified expiration time. For example, `accessTokenMaxAgeSeconds: 17200` would cause the token to expire 48 hours after being issued.
```
Comment (stuartchuan) [inline architecture/additional_concepts/other_api_objects.adoc]: 172800 should be stand for 48 hours

**Comment by mburke5678** [comment]
@stuartchuan Thank you for the correction. Made the change.

**Comment by stuartchuan** [comment]
@mburke5678 Lgtm, thanks.

**Comment by mburke5678** [comment]
@openshift/team-documentation PTAL

**Comment by mburke5678** [comment]
@ahardin-rh Can you take a look? Fairly small change.

File: architecture/additional_concepts/other_api_objects.adoc [hunk 4]
```diff
@@ -111,30 +111,43 @@ The following `*OAuthClient*` objects are automatically created:
...
+<5> One or more absolute URIs can be placed in the `redirectURIs` section. The
 `redirect_uri` parameter sent with authorization requests must be prefixed by
 one of the specified `redirectURIs`.
 ====
 
+[[accessTokenMaxAgeSeconds]]
+The `accessTokenMaxAgeSeconds` value overrides the 
+xref:../../install_config/configuring_authentication.adoc#token-options[default `accessTokenMaxAgeSeconds` value] in the master configuration file 
+for individual OAuth clients. Setting this value for a client allows long-lived access tokens for that client 
+without affecting the lifetime for other clients.
```
Comment (ahardin-rh) [inline architecture/additional_concepts/other_api_objects.adoc]: s/for other clients/of other clients   (?maybe?)

File: install_config/configuring_authentication.adoc [hunk 5]
```diff
@@ -1459,6 +1459,11 @@ The default lifetime is 24 hours.
 codes. The default lifetime is five minutes.
 ====
 
+[NOTE]
+====
+You can override the `*accessTokenMaxAgeSeconds*` value xref:../architecture/additional_concepts/other_api_objects.adoc#oauthclient[through an `*OAuthClient*` Object Definition].
+====
```
Comment (ahardin-rh) [inline install_config/configuring_authentication.adoc]: s/Object Definition/object definition

**Comment by ahardin-rh** [comment]
@mburke Only a few minor suggestions from me. Looks good! 🥂

**Comment by mburke5678** [comment]
[rev_history]
|xref:../architecture/additional_concepts/other_api_objects.adoc#architecture-additional-concepts-other-api-objects[Architecture -> Other API Objects]
|Added new feature allowing xref:../architecture/additional_concepts/other_api_objects.adoc#oauthclient[long-lived access tokens for a specific client] without affecting the lifetime for other clients.
%

**Comment by gaurav-nelson** [comment]
@mburke5678 Is this also applicable to dedicated and online branches? Can you please check?

**Comment by mburke5678** [comment]
@liggitt @stuartchuan Is this feature included in Dedicated or Online?

**Comment by stuartchuan** [comment]
@mburke5678 I think Online will not apply this feature, the Dedicated could use this feature, but need Ops modify oauthClient or create a new oauthClient.

**Comment by mburke5678** [comment]
@stuartchuan @gaurav-nelson 
Should we add the feature into the Dedicated docs with a statement like "If you want to use this feature, please open a support case on the Red Hat Customer Portal."

http://file.rdu.redhat.com/mburke/tokenmaxage-dedicated/architecture/additional_concepts/other_api_objects.html#oauthclient

**Comment by stuartchuan** [comment]
@mburke5678 yes, should added.

**Comment by mburke5678** [comment]
Added the above note in https://github.com/openshift/openshift-docs/pull/6531

**Pull request merged**